rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java

```diff
@@ -1877,6 +1876,33 @@ public void visitStaticMethodCallExpression(StaticMethodCallExpression call) {
             queue.add(new J.MethodInvocation(randomId(), fmt, markers, null, null, name, args, methodType));
         }
 
+        @Override
+        public void visitMethodPointerExpression(MethodPointerExpression ref) {
+            boolean isMethodRef = ref instanceof MethodReferenceExpression;
+            Space fmt = whitespace();
+
+            String referenceName = null;
+            if (ref.getMethodName() instanceof ConstantExpression) {
+                referenceName = ((ConstantExpression) ref.getMethodName()).getValue().toString();
+            }
+
+            queue.add(new J.MemberReference(randomId(),
+                    fmt,
+                    isMethodRef ? Markers.EMPTY : Markers.build(singleton(new MethodPointer(randomId()))),
```

Comment:
While I agree with the change this addition of a new marker likely makes this a core change that needs further review.

Reply:
We could also break it up in two PRs. So this PR would only provide support for method references, and the other (still to be created) PR which add support method pointers with the new marker.